### ISSUE TITLE
[request]: GD support for image generation

### ISSUE THREAD

**Issue by elnath78** [Bacon/BaconQrCode#137]
I'm not sure ifthis was already discussed, but I find much more practical to have GD support than Imagick, iirc GD is natively present in recent PHP intallations, while Imagick must be installed separately, configured and you need to pray it works.

**Comment by DASPRiD** [comment]
There is an open pull requests for an alternate `GdLibRenderer`, but that one hasn't been processed yet:
https://github.com/Bacon/BaconQrCode/pull/114

**Comment by arxeiss** [comment]
@elnath78 as noted above, there is open PR for that. If you want, add this to your `composer.json`.
It works, I'm using it on production for one client. And trying to keep up to date with latest changes.

P.S. There is another reason not to use Imagick. When QR code is too large it was failing for me locally with some kind a weird exception that the path was too complex.

```json
"repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/arxeiss/BaconQrCode"
        }
    ],
    "require": {
        ...
        "bacon/bacon-qr-code": "dev-add-gd-backend#650e10121f4d8707ecdc5984b042cae7731e388a",
```

**Comment by elnath78** [comment]
@arxeiss I tried the Bacon path but it is incredibly slow, tested on local and it averaged 1.76 seconds to generate a QR code, I tested the various available formats. Eventually I decided to use the Google service for QR Code generation, and although it does a remote connection, it is super fast, the average was 0.08 seconds. The only concern is about GDPR, the other possible solution would be generate QR Codes then serve them as users join and request them, possibly an async service that generate a QR Code when the user signs up and serve it ready when he decides to switch to 2FA login.

**Comment by arxeiss** [comment]
How big your QR code is?
I tested it locally:
- Correction level is M 
- Input text is 140 chars length. 
- Final QR is 350x350 px and grid is 40x40 dots
- GD (png) and Imagick (png) is ~80ms to draw with uniform colors.
- Imagick with Gradient is ~180ms.

**Comment by DASPRiD** [comment]
Resolved by #171
